application/apps/indexer/parsers/src/someip.rs

```diff
@@ -250,68 +279,84 @@ fn option_string(option: &SdEndpointOption) -> String {
-                                    | MessageType::Notification => method.get_request(),
-                                    MessageType::Response => method.get_response(),
-                                    _ => None,
-                                }
-                            })
-                        });
-
-                let som_type = fibex_type.and_then(|value| FibexTypes::build(value).ok());
-
-                format!(
-                    "{}::{} {}",
-                    service_name.unwrap_or("Service?"),
-                    method_name.unwrap_or("Method?"),
-                    match payload.is_empty() {
-                        true => Cow::Borrowed(""),
-                        false => {
-                            let mut som_parser = SOMParser::new(payload);
-                            som_type
-                                .map(|mut value| match value.parse(&mut som_parser) {
-                                    Ok(_) => {
-                                        Cow::Owned(format!("{value}").replace([' ', '\n'], ""))
+                match meta_data.model.get_service(service_id, service_version) {
+                    Some(service) => match service.get_method(method_id) {
+                        Some(method) => {
+                            let fibex_type = match message_type {
+                                MessageType::Request
+                                | MessageType::RequestNoReturn
+                                | MessageType::Notification => method.get_request(),
+                                MessageType::Response => method.get_response(),
+                                _ => None,
+                            };
+
+                            let som_type =
+                                fibex_type.and_then(|value| FibexTypes::build(value).ok());
+
+                            format!(
+                                "{}::{} {}",
+                                service.name,
+                                method.name,
+                                match payload.is_empty() {
+                                    true => Cow::Borrowed(""),
+                                    false => {
+                                        let mut som_parser = SOMParser::new(payload);
+                                        som_type
+                                            .map(|mut value| match value.parse(&mut som_parser) {
+                                                Ok(_) => Cow::Owned(
+                                                    format!("{value}").replace([' ', '\n'], ""),
+                                                ),
+                                                Err(error) => {
+                                                    format!("'{}' {:02X?}", error, *payload).into()
+                                                }
+                                            })
+                                            .unwrap_or(format!("{:02X?}", *payload).into())
```

Comment:
`unwrap_or_else()` would be more suitable here to evaluate the inner value only if needed
```rust
unwrap_or_else(|| format!("{:02X?}", *payload).into())
```